### PR TITLE
Android CI

### PR DESCRIPTION
**Pull request by daquexian**
**Description**: 

* Update ONNX for https://github.com/onnx/onnx/pull/2221, which fix segfault on Android

* Some misc changes to pass Android CI

Currently, the yaml file of Android CI is tools/ci_build/github/azure-pipelines/android-x86_64-crosscompile-ci-pipeline.yml

### CODE REVIEW

File: cgmanifest.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
          "component":{
             "type":"git",
             "git": {
-              "commitHash": "65b8e0f9979fbade16e3becbdfa69c0764946f72",
+              "commitHash": "a25e13886e00eed06620d9d59e1ac1eaa18b25b7",
```
Comment (snnn) [inline cgmanifest.json]: Why do you change this?

File: onnxruntime/test/framework/test_tensor_loader.cc [hunk 2]
```diff
@@ -134,7 +134,7 @@ TEST(CApiTest, load_float_tensor_with_external_data) {
   run_external_data_test<false>();
 }
 
-#if defined(__amd64__) || defined(_M_X64)
+#if (defined(__amd64__) || defined(_M_X64)) && not defined(__ANDROID__)
```
Comment (snnn) [inline onnxruntime/test/framework/test_tensor_loader.cc]: && not?

File: tools/ci_build/build.py [hunk 3]
```diff
@@ -550,6 +550,10 @@ def run_onnxruntime_tests(args, source_dir, ctest_path, build_dir, configs, enab
             run_subprocess(os.path.join(source_dir, 'tools', 'ci_build', 'github', 'android', 'start_android_emulator.sh'))
             adb_push(source_dir, 'testdata', '/data/local/tmp/', cwd=cwd)
             adb_push(source_dir, os.path.join(source_dir, 'cmake', 'external', 'onnx', 'onnx', 'backend', 'test'), '/data/local/tmp/', cwd=cwd)
+            # Disable scan op test for nnapi ep
+            if enable_nnapi:
+                adb_shell('rm', '-rf', '/data/local/tmp/test/data/node/test_scan9_sum',
```
Comment (snnn) [inline tools/ci_build/build.py]: If you want to ignore this test, you may add it to onnx_test_runner's main.cc

File: cgmanifest.json (hunk 1 above)
Comment (linkerzhang) [reply cgmanifest.json]: This is what's in the description "Update ONNX for onnx/onnx#2221, which fix segfault on Android"

but, @daquexian do you know the root cause in ONNX which leads to the segfault please? if yes, could you share more details about it please? Thank you!

Comment (daquexian) [reply cgmanifest.json]: @snnn @linkerzhang It is a bug in onnx. `ctx.getInputType(1)` (https://github.com/onnx/onnx/pull/2221/files#diff-f24e7f9545820de198f21ac7c7d37d10R1542) is not always non-null. It causes `onnx_test_runner` crashes on pytorch repeat op in the test data.

File: tools/ci_build/build.py (hunk 3 above)
Comment (daquexian) [reply tools/ci_build/build.py]: Thanks! I have updated it.

**Comment by daquexian** [comment]
@linkerzhang @snnn could you please run the CI? BTW, is Android CI enabled now?

**Comment by daquexian** [comment]
@snnn Could you please trigger the CI again? I have fixed windows ci

File: onnxruntime/test/framework/test_tensor_loader.cc [hunk 4]
```diff
@@ -134,7 +134,8 @@ TEST(CApiTest, load_float_tensor_with_external_data) {
   run_external_data_test<false>();
 }
 
-#if (defined(__amd64__) || defined(_M_X64)) && not defined(__ANDROID__)
+#if defined(__amd64__) || defined(_M_X64)
+#ifndef __ANDROID__
```
[the remark this answers is not in the document]
Comment (daquexian) [reply onnxruntime/test/framework/test_tensor_loader.cc]: Android can run on x86 and x64 processors, e.g., android x64 emulators

**Comment by daquexian** [comment]
@snnn @linkerzhang Build passed but CI failed due to an error "The Component Detection tool experienced a failure. See the logs for more information"

**Comment by daquexian** [comment]
@linkerzhang @snnn @shahasad Android CI works now

**Comment by shahasad** [comment]
The Android CI fails on the code (when master is merged)
https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=74923

**Comment by snnn** [comment]
@shahasad Pranav will fix the test error.

**Pull request merged**